src/Post.js

```diff
@@ -37,23 +35,32 @@ class Post extends React.Component {
       .then(this.props.loadPosts)
   }
 
-  handleClick(e) {
+  loadEditForm(e) {
     e.preventDefault();
-    this.setState({
-      editing: true
-    })
-    this.show()
+    this.setState({ editing: true })
   }
 
-  handleSubmit(event) {
-    event.preventDefault();
-    let data = {
-      message: this.state.updated_post
+  fetchParams() {
```

Comment:
this could be a private function

Reply:
also basically the same as new post fetchParams function, is there a way to just have one function and pass argument of the value of message: in data?

Reply:
That probably is the dryest way, but there's enough changes that it gets pretty fiddly. method, credentials, body vs no body, cors. I think you're right but I'm going to leave it as is for now